### PR TITLE
Updated to work with Xcode 10.2

### PR DESCRIPTION
**Pull request by mohpor**

### CODE REVIEW

File: Source/MatchResult.swift [hunk 1]
```diff
@@ -67,17 +67,17 @@ public struct MatchResult {
   internal init(_ string: String, _ result: NSTextCheckingResult) {
     self._result = _MatchResult(string, result)
     self._string = string
-    self._captureRanges = Memo { [_result] in
+    self._captureRanges = Memo(unevaluated: { [_result] in
```
[the remark this answers is not in the document]
Comment (mohpor) [reply Source/MatchResult.swift:70]: This is something I've had to add because compiler was getting confused by the identical initializers.

**Review by cristipetra — APPROVED** [review]

File: Source/MatchResult.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (dohyeondk) [reply Source/MatchResult.swift:70]: @mohpor +1

We could rename private instance variables without `_` since Swift avoids it.

**Review by manoj036 — APPROVED** [review]

**Comment by manoj036** [comment]
@sharplet Can you merge this one. I need this pull request soon. Thanks

**Comment by sharplet** [comment]
Thanks @mohpor! I ended up merging https://github.com/sharplet/Regex/pull/63 in order to retain 4.2 support. I’m pushing 2.0 shortly, let me know if there’s any issues.

**Pull request closed without merging**